src/main/java/com/spartronics4915/lib/lidar/LidarProcessor.java

```diff
@@ -339,18 +350,19 @@ public void addPoint(double ts, double angle, double dist,
         Pose2d robotLoc = null;
         if(this.mMode == OperatingMode.kAbsolute)
         {
-            IRobotStateMap.State robotState = null;
-            if (sRobotStateMap.containsKey(ts)) 
+            Pose2d robotPose = null;
+            if (sLidarPoseCache.containsKey(ts)) 
             {
-                robotState = sRobotStateMap.get(ts);
+                robotPose = sLidarPoseCache.get(ts);
             } 
             else
             {
-                robotState = sRobotStateMap.get(ts);
-                if(robotState != null)
-                    sRobotStateMap.put(ts, robotState);
+                // This is to correct for time smear
+                robotPose = mEncoderStateMap.getFieldToVehicle(ts).transformBy(kVehicleToLidar);
+                if(robotPose != null)
+                    sLidarPoseCache.put(ts, robotPose);
```

Comment:
@dbadb Before this change we were putting null back in when we got null. Are there any nuances that I was missing? (I just assumed that this was a bug and fixed it).